pkg/manifests/manifests.go

```diff
@@ -1424,6 +1425,29 @@ func (f *Factory) PrometheusK8s(grpcTLS *v1.Secret, telemetrySecret *v1.Secret)
 	for k := range p.Spec.RemoteWrite {
 		rw := &p.Spec.RemoteWrite[k]
 
+		// Fixes: https://issues.redhat.com/browse/OCPBUGS-38289
+		// The code is condensed here to simplify backporting.
+		// This is a temporary workaround because the NoProxy field isn't exported on the Prometheus CRD in older versions.
+		// If the cluster proxy or remoteWrite config change, this code will re-run, ensuring no stale state.
+		// The httpproxy package used here is the same as used by Prometheus, which helps with consistency.
+		if f.proxy.HTTPSProxy() == "" && f.proxy.HTTPProxy() == "" {
+			continue
+		}
+		// Check if the endpoint URL needs proxying.
+		proxyConfig := &httpproxy.Config{
+			HTTPProxy:  f.proxy.HTTPProxy(),
+			HTTPSProxy: f.proxy.HTTPSProxy(),
+			NoProxy:    f.proxy.NoProxy(),
+		}
+		proxyFunc := proxyConfig.ProxyFunc()
+		// Ignore errors here.
+		u, _ := url.Parse(rw.URL)
```

Comment:
An improvement for later (and it can wait for the CMO config CRD) but we could detect invalid URLs and fail early. I suppose that right now, a not parseable URL would fail the prometheus operator reconciliation.

Reply:
Yes, we just need to make sure they use the same definition of a valid URL.
I thought that it would be blocked by the validation webhook, but I'm not sure.

Reply:
which validation webhook?

Reply:
prom-operator's, but maybe Prometheus CRs do not go through it.

Reply:
yeah, the validation webhook only works against PrometheusRule and AlertmanagerConfig objects.